great_expectations/data_context/data_context.py

```diff
@@ -385,9 +386,7 @@ def __init__(
         self._data_context_id = self._construct_data_context_id()
 
         # Override the project_config data_context_id if an expectations_store was already set up
-        self._project_config.anonymous_usage_statistics.data_context_id = (
-            self._data_context_id
-        )
+        self.config.anonymous_usage_statistics.data_context_id = self._data_context_id
```

Comment:
I realize this behavior was already here but why are we mutating the config object after assignment? I'd feel more secure if the config attribute was immutable.

Reply:
@cdkini This is a very good question (and there is more to say about it).  The short answer is that one can use Python to add components (e.g., add another store) programmatically, and save the updated configuration, and persist it to a YAML file.  So we should remove "immutable" throughout, because it is not.  I will do it and commit in this PR.  Thanks.

Reply:
I think the config should be immutable though? 

We're using config objects to create business logic objects and saving those config as attributes - it feels a bit iffy to me to be changing the value of the config attribute after the config object's usage.